runtime/sema/check_return_statement.go

```diff
@@ -30,48 +30,67 @@ func (checker *Checker) VisitReturnStatement(statement *ast.ReturnStatement) ast
 		functionActivation.ReturnInfo.DefinitelyReturned = true
 	}()
 
-	// check value type matches enclosing function's return type
+	returnType := functionActivation.ReturnType
 
 	if statement.Expression == nil {
+
+		// If the return statement has no expression,
+		// and the enclosing function's return type is Void,
```

Comment:
```suggestion
		// and the enclosing function's return type is non-Void,
```

Is this what you meant?

Reply:
Yes, good catch!